src/symbols.c

```diff
@@ -324,6 +324,56 @@ static gint compare_symbol_lines(gconstpointer a, gconstpointer b)
 }
 
 
+static gboolean is_symbol_within_parent(const TMTag *tag, const TMTag *parent)
+{
+	const gchar *scope_sep = tm_parser_scope_separator(tag->lang);
+	guint scope_len = 0;
+
+	if (EMPTY(tag->scope))
+		return FALSE;
+
+	if (!EMPTY(parent->scope))
+	{
+		if (!g_str_has_prefix(tag->scope, parent->scope))
+			return FALSE;
+		scope_len = strlen(parent->scope);
+
+		if (!g_str_has_prefix(tag->scope + scope_len, scope_sep))
+			return FALSE;
+		scope_len += strlen(scope_sep);
+	}
+
+	if (!g_str_has_prefix(tag->scope + scope_len, parent->name))
+		return FALSE;
+	scope_len += strlen(parent->name);
+
+	if (scope_len == strlen(tag->scope))
```

Comment:
Nitpick: I'd rather write `! tag->scope[scope_len]` or one of the many equivalents that avoid yet another `strlen()`

Reply:
Yes, this one crossed my mind before falling asleep yesterday :-)